### ISSUE TITLE
Incorrect warning with ucx master

### ISSUE THREAD

**Issue by hjelmn** [openucx/ucx#2953]
Seeing this when I set ```UCX_NET_DEVICES```:

```
export UCX_NET_DEVICES=mlx5_0:1,mlx5_3:1
...
[1539371008.388533] [cn828:74988:0]         parser.c:1310 UCX  WARN  unused env variable: UCX_NET_DEVICES (set UCX_WARN_UNUSED_ENV_VARS=n to suppress this warning)
```

If I set something that doesn't exist in that variable I get the expected warning so it **is** being used:

```
export UCX_NET_DEVICES=self,mlx5_0:1,mlx5_3:1
...
[1539370849.809991] [cn828:74750:0]    ucp_context.c:588  UCX  WARN  device 'self' is not available
[1539370849.810202] [cn828:74750:0]         parser.c:1310 UCX  WARN  unused env variable: UCX_NET_DEVICES (set UCX_WARN_UNUSED_ENV_VARS=n to suppress this warning)
```

**Comment by yosefe** [comment]
command line?

**Comment by hjelmn** [comment]
Got it with both of these:

```mpirun -n 2 -N 1 --mca pml ucx ./NPmpi```
```mpirun -n 2 -N 1 --mca pml ob1 --mca btl self,vader,uct --mca btl_uct_memory_domains ib/mlx5_0 ./NPmpi```

**Comment by yosefe** [comment]
can you try passing "-x UCX_NET_DEVICES=mlx5_0:1,mlx5_3:1" to mpirun instead of "export"?
maybe the warning comes from "mpirun" process

**Comment by AdamSimpson** [comment]
I've noticed this same issue regardless of how I set the variable.  It happens with the v1.4.x branch but not with master.

**Comment by hjelmn** [comment]
Actually, this might be v1.4.0 for me as well. I need to see if it was using the system one (v1.4.0) or my build (master).  Its most likely the later as v1.4.0 crashes with multi-threaded codes on aarch64 (the system tested).

**Comment by yosefe** [comment]
fix: #2972

**Comment by yosefe** [comment]
@hjelmn @AdamSimpson FYI should be fixed now with UCX master
